src/main/java/com/teragrep/pth_06/FileSystemFactoryImpl.java

```diff
@@ -0,0 +1,181 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06;
+
+import com.teragrep.pth_06.config.HdfsConfig;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+
+public final class FileSystemFactoryImpl implements FileSystemFactory {
+
+    final Logger LOGGER = LoggerFactory.getLogger(FileSystemFactoryImpl.class);
+
+    private final String kerberosAuthentication;
+    private final String hdfsUri;
+    private final String kerberosRealm;
+    private final String kerberosKdc;
+    private final String kerberosAuthorization;
+    private final String kerberosAutorenewal;
+    private final String UseHdfsHostname;
+    private final String kerberosPrincipalPattern;
+    private final String hdfsTransferProtection;
+    private final String hdfsCipherSuites;
+    private final String kerberosKeytabUser;
+    private final String kerberosKeytabPath;
+
+    public FileSystemFactoryImpl(HdfsConfig config) {
+        this(
```

Comment:
why not use `HdfsConfig` as a field instead of separating them like this?

Reply:
Spark serializes all the inputpartitions, which means that HdfsConfig that is not serializable can't be set as field in HdfsMicroBatchInputPartition.java and instead it must be separated like this,, so it can be also used in HdfsMicroBatchInputPartitionReader.